tests/integration/collection/update/simple/with_filter_test.go

```diff
@@ -13,16 +13,41 @@ package update
 import (
 	"testing"
 
+	"github.com/sourcenetwork/immutable"
+
 	testUtils "github.com/sourcenetwork/defradb/tests/integration"
 )
 
 func TestUpdateWithInvalidFilterType_ReturnsError(t *testing.T) {
+	type invalidFilterType struct{ Number int }
 	test := testUtils.TestCase{
 		Description: "Test update users with invalid filter type",
+		// http and cli clients will pass the serialize filter into json which will result in
+		// the payload deserialized into map[string]any. With Go client the filter is passed as is.
+		SupportedClientTypes: immutable.Some(
+			[]testUtils.ClientType{testUtils.HTTPClientType, testUtils.CLIClientType}),
+		Actions: []any{
+			testUtils.UpdateWithFilter{
```

Comment:
praise: I forgot we had this, thanks for using it and not doing whatever I suggested in the standup :grin:  Thanks for the comment too, it explains it nicely.